libs/qlik/src/dashboard/vl-qlik-dashboard.component.ts

```diff
@@ -0,0 +1,369 @@
+import { css, CSSResult, html, PropertyDeclarations } from 'lit';
+import { BaseLitElement, webComponent } from '@domg-wc/common-utilities';
+import { stardust } from '@nebula.js/stardust/types';
+import { Qlik, STARDUST } from '@domg/qlik-lib';
+import { bindVlSelect, debounce, queryById, queryBySelectorAll, renderStack } from '../utils/qlik-render-helpers';
+import { DashboardVisualization, Filter } from '../utils/vl-qlik.types';
+
+import '../visual';
+import { vlElementsStyle } from '@domg-wc/elements';
+
+@webComponent('vl-qlik-dashboard')
+export class VlQlikDashboardComponent extends BaseLitElement {
+    private connected = false;
+    private initialized = false;
+    private filtersLoading = true;
+    public visuals: Array<Array<DashboardVisualization>>;
+    public filters: Array<Filter> = [];
+    public connection?: Qlik;
+    private stardust?: stardust.Embed;
+    private selected = {};
+
+    static get styles(): (CSSResult | CSSResult[])[] {
+        return [vlElementsStyle];
+    }
+
+    static get properties(): PropertyDeclarations {
+        return {
+            visuals: { type: Array },
+            filters: { type: Array },
+            connection: { type: Object },
+            connected: { type: Boolean, state: true },
+            initialized: { type: Boolean, state: true },
+            filtersLoading: { type: Boolean, state: true },
+            selected: { type: Object, state: true },
+        };
+    }
+
+    async firstUpdated() {
+        if (this.connection && !this.connected) {
+            this.stardust = await STARDUST(this.connection.app);
+            this.connected = true;
+            this.dispatchEvent(new CustomEvent('connected'));
+        }
+        if (this.connected && this.filters && this.filters.length > 0) {
+            try {
+                await Promise.all(this.filters.map((f) => this.connection.addFilters(f.filter)));
+            } catch (e) {
+                // do nothing
+            }
+        }
+        this.initialized = true;
+        this.dispatchEvent(new CustomEvent('initialized'));
+        this.search();
+    }
+
+    async updated() {
+        if (this.initialized && this.filters && this.filters.length > 0) {
+            this.__bindFilters();
+        }
+    }
+
+    render() {
+        if (!this.initialized || !this.connected) {
+            return renderStack({
+                size: 12,
+                template: html` <vl-loader data-vl-text="Dashboard is aan het laden"></vl-loader>`,
+            });
+        }
+        return html`
+            <style>
```

Comment:
kan de styling niet in een apart 'vl-qlik-dashboard.uig-css.ts' bestand conform bvb. 'vl-accordion.uig-css.ts' (dat zijn ook typescript bestanden dus ook generatie logica zou daarin kunnen)